### ISSUE TITLE
Question about the experimental results of 1D Burgers

### ISSUE THREAD

**Issue by jianke0604** [tum-pbs/ConFIG#6]
Hello, thanks for your wonderful work! I tried to follow your work and reproduce the results of the Burgers equation in Table 2, but I seem to have encountered some problems.
Specifically, I followed the setup in README.md and ran the following code:
```
run_burgers(
    name="ConFIG_2",
    trainer=get_gradvec_trainer(BurgersTrainerBasis, ConFIGOperator()),
    save_path=SAVE_PATH,
    n_losses=2, 
)
```
I did not change any parameters in the code, but the results of the experiment are as follows:
```
2024-11-20 14:46:05 : There are 1 training batches in each epoch
2024-11-20 14:46:05 : Training epochs:30000
2024-11-20 14:46:05 : Total training iterations:30000
2024-11-20 14:46:05 : learning rate:0.001
2024-11-20 14:46:05 : Optimizer:Adam
2024-11-20 14:46:05 : Learning rate scheduler:cosine
2024-11-20 14:46:05 : Use learning rate warm up, warmup epoch:100
2024-11-20 14:46:05 : Training start!
2024-11-20 14:53:57 : Training finished!
2024-11-20 14:53:57 : Final training loss: 0.0002263441710965708
2024-11-20 14:53:57 : Final validation loss: 0.00016145150584634393
2024-11-20 14:53:57 : Best validation loss: 0.00012893968960270286 at epoch 27057
2024-11-20 14:53:57 : Training losses of the final 500 epochs: min:1.08309e-04, max:3.47662e-04, average:1.62916e-04, median:1.56166e-04
2024-11-20 14:53:57 : Validation losses of the final 500 epochs: min:1.38220e-04, max:1.91180e-04, average:1.59201e-04, median:1.58736e-04
2024-11-20 14:53:57 : Total running time: 00h:07m:56s
2024-11-20 14:53:57 : Total training time: 00h:06m:52s
2024-11-20 14:53:57 : Training speed: 0.01376 s/iteration
```
You can see that the validation loss after training is about 1.58e-4. When I run the test as follows:
```
network=TrainedProject("./PINN_trained/burgers/baseline/2024-05-20-12_43_38/").get_saved_network() #The folder in specificed in training procedure
simulation_data=np.load("./data/burgers/simulation_data.npy")
mse_value,mse,prediction=run_test(network=network,simulation_data=simulation_data,device="cuda:0")
print("MSE: ",mse_value)
```
The final result is also about 1.58e-4. Using PCGrad will produce similar results, which is inconsistent with the result of about 1.3e-4 in Table 2. Could you please give further explanation on the experimental results or experimental setup? Since I don't have a deep understanding of the field of PDE, there may be some misunderstandings. I would be grateful if you can correct me further.

**Comment by qiauil** [comment]
Hi, @jianke0604 

In our manuscript, we choose the min value as the final result, as is shown in your log, which is 1.38220e-4.
The reason we use the min value is that we update the data points during the training. Thus, the training loss could be very noisy. If you want to get a smooth training loss, you can turn off the data update in the configuration or add a grad clip during training. Then, comparing the mean or media performance would be more meaningful.

Cheers

**Comment by qiauil** [comment]
B.T.W., I am not sure which value I choose to show in the manuscript, maybe it is the best value during the whole training, then the value for this training is 1.28e-4.
